File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/PhoneNumberExtract.java
package com.cuupa.classificator.services.kb.semantic.dataExtraction;

import org.jetbrains.annotations.NotNull;

import java.util.regex.Pattern;

public class PhoneNumberExtract extends Extract {

    private final Pattern pattern;

    public PhoneNumberExtract(@NotNull String regex) {
        pattern = Pattern.compile(regex);
    }

    @Override
    public Pattern getPattern() {
        return pattern;
    }

    @Override
    public String normalize(final String value) {
        return value;
    }
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/PhoneNumberExtract.kt
package com.cuupa.classificator.services.kb.semantic.dataExtraction

import java.util.regex.Pattern

class PhoneNumberExtract(regex: String) : Extract(Pattern.compile(regex.trim())) {

    override fun normalize(value: String): String {
        return value
    }

    companion object{
        const val name = "[PHONENUMBER]"
    }
}